src/content/en/fundamentals/instant-and-offline/offline-storage-pwa/index.md

```diff
@@ -0,0 +1,260 @@
+key-value data management
+
+* [Promisified
+IndexedDB](https://github.com/inexorabletash/indexeddb-promises): native
+support for a Promise-friendly version of IDB
+
+* [IndexedDB Observers](https://github.com/WICG/indexed-db-observers): native
+IDB observation without needing wrapper around the database
+
+* [Async Cookies API](https://github.com/bsittler/async-cookies-api): async
+JavaScript cookies API for documents and workers
+
+* [Quota Management API](https://www.w3.org/TR/quota-api/): check how much
+quota an app/origin is using
+
+* [writable-files](https://github.com/WICG/writable-files): allow sites to
+interact with local files more seamlessly
+
+* [Directory downloads](https://github.com/drufball/directory-download): allow
+sites to download directories without .zip files
+
+* [File and Directory Entries API](https://wicg.github.io/entries-api/):
+support for file and directory upload by drag-and-drop.
+
+* Support for an [Async Cookies
+API](https://github.com/WICG/async-cookies-api) is being sketched out right now
+with a polyfill in the works.
+
+* Debugging IDB is not currently supported in Edge (however, it is possible to
+debug the underlying JetDB) — vote
+[here](https://wpdev.uservoice.com/forums/257854-microsoft-edge-developer/suggestions/6517763-indexeddb-explorer-in-dev-tools)
+for built in support.
+
+* Although [ideas](https://github.com/slightlyoff/async-local-storage) for
+async LS have been kicked around in the past, current focus is on getting
+[IndexedDB 2.0](https://w3c.github.io/IndexedDB/) in a good state.
+
+* The [writable-files](https://github.com/WICG/writable-files) proposal may
+eventually give us a better standards-track solution for seamless local file
+interaction.
+
+* For apps requiring more persistent storage, see the on-going work on
+[Durable Storage](https://storage.spec.whatwg.org/).
+
+Offline storage isn’t quite magical and an understanding of the underlying APIs
+will go far in helping you make the most out of what we now have available.
+Whether you prefer to directly use these APIs or work with an abstraction
+library, take some time to get familiar with your options.
+
+Hopefully this guidance will help craft an offline experience that makes your
+PWA shine! ✨
+
+## **Background Reading**
+
+* [State of offline storage
+APIs](https://docs.google.com/presentation/d/11CJnf77N45qPFAhASwnfRNeEMJfR-E_x05v1Z6Rh5HA/edit)
+by Joshua Bell
+
+* [Browser Database
+Comparison](http://nolanlawson.github.io/database-comparison/) by Nolan Lawson
+
+* [IndexedDB, WebSQL, LocalStorage — what blocks the
+DOM?](https://nolanlawson.com/2015/09/29/indexeddb-websql-localstorage-what-blocks-the-dom/)
+
+* [How to think about databases (Pokedex
+research)](https://nolanlawson.com/2016/02/08/how-to-think-about-databases/)
+
+* [Which APIs are supported in Web Workers and Service
+Workers?](https://nolanlawson.github.io/html5workertest/)
+
+*T**hanks to Nolan Lawson, Joshua Bell (whose work on Open Web Storage and
```

Comment:
Some of the formatting on words like 'Thanks' here appear to have been altered by the conversion process. We might want to tweak that.